llama-index-integrations/embeddings/llama-index-embeddings-bedrock/llama_index/embeddings/bedrock/base.py

```diff
@@ -74,6 +75,10 @@ class BedrockEmbedding(BaseEmbedding):
     additional_kwargs: Dict[str, Any] = Field(
         default_factory=dict, description="Additional kwargs for the bedrock client."
     )
+    titan_body_kwargs: Dict[str, Any] = Field(
```

Comment:
it feels a tad weird to add a specific field for a specific embedding model that bedrock supports? I feel like this wouldn't scale well

I feel like additional_kwargs serves this usecase generically? Its currently not even being used lol

Reply:
@logan-markewich additional_kwargs effectively guarantees you have to read the LlamaIndex source code to use a baseline LlamaIndex embedding.

Currently, at least within Bedrock, additional_kwargs is used only top-level fields across the implementations.  What’s the proposal for having fields within sub-fields, which would be consistent across the LlamaIndex integrations?  Would a nested dictionary be appropriate?

I do think LlamaIndex’s Bedrock code needs to become a direct variant of langchain-aws for long-term maintainability, rather than its current state.

Reply:
Hmm sounds like there is some dissonance between our intention for `additional_kwargs` and Bedrock's. Maybe we could add a new field `model_config` or some new base data class? Something like below:

# model_args.py
```python
class ModelConfig[BaseModel]
```

and then subclass it here for Titan with a field for `body_kwargs`.

```python
class TitanModelConfig[ModelConfig]:
    body_kwargs: ...
```

And use it as such:

```
model_config: Optional[ModelConfig] = Field(default=None, ...)
```

Reply:
I'm personally not a fan of model config classes since they often contribute to code bloat. This would also be the only (current) instance in which those are used within LlamaIndex.

The feedback @logan-markewich provided led to an overall improvement, while still just relying on `additional_kwargs` ([here](https://github.com/run-llama/llama_index/pull/13580/commits/30363258d995528ba7a6cf2d11540bd5534b5f0f)).

Reply:
Sgtm!